spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/TaskService.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.server.service;
+
+import java.util.Map;
+
+/**
+ * Provides Task related services.
+ *
+ * @author Michael Minella
+ * @author Marius Bogoevici
+ * @author Glenn Renfro
+ * @author Mark Fisher
+ * @author Janne Valkealahti
+ * @author Gunnar Hillert
+ *
+ */
+public interface TaskService {
+
+	/**
+	 * Execute a task with the provided task name and optional runtime properties.
+	 *
+	 * @param taskName Name of the task. Must not be null or empty.
+	 * @param runtimeProperties Optional runtime properties. Must not be null.
+	 */
+	void executeTask(String taskName, Map<String, String> runtimeProperties);
```

Comment:
Looking for consistency at all levels here.  Whether we call them properties or params is fielders choice.   So long as they are not confused with the "properties" that are needed by the deployer.